rfcs/045-sierra-work-relationships/README.md

```diff
@@ -0,0 +1,377 @@
+contains a $w subfield, then 
+__Investigate where this happens in order to decide, before developing a solution__
+ * Find out if it ever happens
+ * Find out how many objects there are in each "series" where it does happen
+ * Ask C&R what it means
+ 
+### subPart property name
+
+The subfields `$g - Related parts` and `$v Volume/Sequential designation` are used in the
+Sierra data in 773/774 and 440/490/830 fields, respectively.  
+Giving them a commonname in the API is simpler than
+having two different names, forcing a client to first try one then the other
+or look at one property if the parent is a Series and another one if the
+parent is a Work or Unknown.
+
+### No Nesting of part/partOf
+
+Whereas hierarchies from CALM data can be arbitrarily deep and broad, 
+and benefit from nested `partOf` and `part` values, 
+values derived from these fields are effectively flat. Although the 
+first layer in either may be arbitrarily broad (multiple parents/children), 
+the next layer would have maximally one value.
+
+In addition, where `$g` or `$v` are used, the immediate parent/child as proposed in
+the original RFC is of less importance, being a volume number or page number.
+
+They denote membership of a container, and optionally denote a subPart
+within that container.
+
+Semantically, it may be correct to define an object by nesting, but in
+practice, using a structure that allows arbitrarily deep and wide nesting
+adds unnecessary complexity to both server and client.
+
+For each field value from Sierra creating a partOf object
+
+* There is optionally one grandparent
+* The terminal ancestor is the "important" one.
+
+And the same is true In a part relationship derived from these fields:
+
+* Each child optionally has one child
+* The terminal descendant is the "important" one.
+
+Using nesting in this situation would mean that a client would have to 
+iterate to find terminal ancestors/descendants.  This pattern may be clear to 
+Wellcome Collection developers, but not obvious to external consumers of the API.
+
+## Linking a Work to a Series
+
+### Manifestation
+
+These *are not* to be presented within a hierarchy. On the website, 
+the name of the series should present as a link to a filtered search
+for other objects in the same series.  The behaviour of the API will need 
+to be updated in order to support filtering partOf by `title` as well as `id`
+as it currently does.
+
+In the API, this should be represented in a partOf value with no id, and a 
+type of "Series", thus:
+
+```json
+"partOf": [
+  {
+    "title": "Fallaize Collection.",
+    "type": "Series"
+  }
+]
+```
+
+A MARC value may have a subfield denoting a "part" of the series, this should
+be separated from the series title, and presented in a new field, `subPart`
+
+In the case of 830, 430 nd 490 fields, this is in the $v subField.
+
+`830 Published papers (Wellcome Chemical Research Laboratories)
```

Comment:
This example demonstrates a key aspect of MARC, that subfields are generally written to be concatenated in order and contain the punctuation required to do so.

That means in this instance we'd really need to strip the ` ;`. However, past experience shows we need to be a bit careful doing that, as it's not always so straightforward. We'd also then have to consider how we display the separate parts when we join them back together. And if we are just going to join them back together, it may make sense not to split them in the first place...

Most of the times we've split in the past, including `title` and `physicalDescription`, we've ended up backing it out and just combining the MARC as intended. I think given that, there's a strong argument to start with the combined title, then look again at this as a potential improvement if the shape of the data shows that it's necessary.

Reply:
OK.  So we could say 

Step 1: include the whole title including subfields
Step 4: crack the title into its constituent subfields
Step 5: work out what to do with the subfields as distinct from the main title.

Doing this would not give us the advantage described here: https://github.com/wellcomecollection/docs/blob/4cb54a95884a8b0821ceb0f73b7284e8976d1fbb/rfcs/045-sierra-work-relationships/README.md?plain=1#L239-L249 until we do crack the subfields, but we would reach the MVP quicker as we wouldn't have to work out those punctuation kinks early on.

As it links to a search, the likely User Journey would be:
0. From the record - 
1. Click on the Series Link e.g. "Published papers (Wellcome Chemical Research Laboratories) ; no. 149"
2. See the search page with one item on it (the one we have just come from)
3. manually remove the suffix from the search query and search again

However, this raises UI issue.  This should ideally result in a search filtered on a specific field, rather than full text, and all our existing field-based filters are checkboxes.  To facilitate this, we would have to allow free-text fielded search.

Perhaps the first step then is to link to a full-text search for the series title, then later refine this to search in the appropriate field.

Reply:
Drafting this response, I discovered this bug. https://github.com/wellcomecollection/platform/issues/5436

Reply:
Following discussions here: https://wellcome.slack.com/archives/C3TQSF63C/p1645440176157099

It looks like we have to split the title from the beginning.

Reply:
I have added a section in Unanswered Questions about working out how to deal with subfield separator punctuation.

Reply:
I see this as a "Series" filter, that does a keyword match on the title of the series (whether that includes the volume tbd). That will then appear in the "applied filters" section of the search bar, not as text in the search field.

Does look like we'll want that to just be on the series title, not the volume, making the partNumber/partName something we should probably do.